pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -770,6 +772,40 @@ type SNSConfig struct {
 	HTTPConfig *HTTPConfig `json:"httpConfig,omitempty"`
 }
 
+// TelegramConfig configures notifications via Telegram.
+// See https://prometheus.io/docs/alerting/latest/configuration/#telegram_config
+type TelegramConfig struct {
+	// Whether to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// The Telegram API URL i.e. https://api.telegram.org.
+	// If not specified, default API URL will be used.
+	// +optional
+	APIURL string `json:"apiURL,omitempty"`
+	// Telegram bot token
+	// The secret needs to be in the same namespace as the AlertmanagerConfig
+	// object and accessible by the Prometheus Operator.
+	// +kubebuilder:validation:Required
+	BotToken *v1.SecretKeySelector `json:"botToken,omitempty"`
+	// The Telegram chat ID.
+	// The secret needs to be in the same namespace as the AlertmanagerConfig
+	// object and accessible by the Prometheus Operator.
+	// +kubebuilder:validation:Required
+	ChatID *v1.SecretKeySelector `json:"chatID,omitempty"`
```

Comment:
Does this need to be kept in a Secret? I ask because looking at https://github.com/prometheus/alertmanager/blob/a6d10bd5bc3f651e0ca04d47b981ed66e85a09a6/config/notifiers.go#L644-L645 it appears not.

Reply:
You are right here too! ChatID should be just a number. I'll address it :)